packages/protocol/src/action-protocol/element-type-hints.ts

```diff
@@ -73,9 +74,18 @@ export interface EdgeTypeHint extends TypeHint {
     readonly sourceElementTypeIds: string[];
 
     /**
-     * Allowed targe element types for this edge type
+     * Allowed target element types for this edge type
      */
     readonly targetElementTypeIds: string[];
+
+    /**
+     * Indicates whether this type hint is dynamic or not. Dynamic edge type hints
+     * require an additional runtime check before creating an edge, when checking
+     * source and target element types is not sufficient.
+     *
+     * @see {@link RequestCheckEdgeTargetAction}
+     */
+    readonly dynamic?: boolean;
```

Comment:
Just to make sure I got this right: So we basically now support three main usecases,right?:
-  Allowed target detection via a set of allowed source/target ids that are queried one time during startup
-  Mixed. i.e. first detect via allowed source/target id array and then trigger the dynamic server check
- Pure dynamic. I.e. the source/target id arrays are empty and we forward every target check to the server

Reply:
No, there are only 2 cases supported: fixed and mixed.

To define a Pure dynamic case, you'd need to define an edge that accepts all possible source and target types, and make it dynamic. The static check (sourceType/targetType) is always done before the dynamic check, so if that fails, the dynamic check is skipped entirely.